src/components/user-defined-sets/CurrentSetManager.js

```diff
@@ -35,73 +37,51 @@ export default class CurrentSetManager extends React.Component {
 
   handleSubmit(event) {
     event.preventDefault();
-    const { setName } = this.state;
+    const { isEditing, setName } = this.state;
     const { sets, onUpdateSets } = this.props;
-    onUpdateSets(Sets.nameCurrentSet(sets, setName, false));
+    onUpdateSets(Sets.nameCurrentSet(sets, (isEditing ? setName : this.getNextSetName()), false));
     this.stopEditing();
   }
 
-  // eslint-disable-next-line class-methods-use-this
-  Disabled() {
-    return (
-      <tr>
-        <td className="set-name">{CURRENT_SELECTION}</td>
-        <td />
-        <td />
-      </tr>
-    );
-  }
-
-  EnabledStatic() {
-    return (
-      <tr>
-        <td className="set-name">{CURRENT_SELECTION}</td>
-        <td />
-        <td>
-          <button type="button" className="set-item-button set-item-save" onClick={this.startEditing}>Save</button>
-        </td>
-      </tr>
-    );
-  }
-
-  EnabledEditing() {
-    const { setName } = this.state;
-    return (
-      <tr>
-        <td className="set-name">
-          <input type="text" value={setName} onChange={this.handleChange} />
-        </td>
-        <td>
-          <button type="button" className="set-item-button set-item-cancel" onClick={this.stopEditing}>×</button>
-        </td>
-        <td>
-          <button type="submit" className="set-item-button set-item-save">Save</button>
-        </td>
-      </tr>
-    );
-  }
-
   render() {
     const { sets } = this.props;
-    const { isEditing } = this.state;
 
-    if (!sets || sets.currentSet.size === 0) {
+    if (!sets || sets.currentSet.length === 0) {
       return (
         <table className="current-set-manager sets-manager-disabled">
           <tbody>
-            {this.Disabled()}
+            <tr>
+              <td className="set-name">
+                    No current selection
+              </td>
+              <td />
+              <td />
+            </tr>
           </tbody>
         </table>
       );
     }
+
+    const { setName, isEditing } = this.state;
+
+    const inputProps = {
+      value: (isEditing ? setName : this.getNextSetName()),
+      onChange: (isEditing ? this.handleChange : () => {}),
+      onFocus: (!isEditing ? this.startEditing : undefined),
```

Comment:
```suggestion
      onFocus: (isEditing ? undefined : this.startEditing),
```
Keep the order consistent?